test-cypress/integration/repository/ontop-repository.spec.js

```diff
@@ -152,6 +239,131 @@ describe('Ontop repositories', () => {
 
         // Then I expect url to be calculated properly
         OntopRepositorySteps.getUrlInput().should('have.value', 'jdbc:snowflake://someHostName.snowflakecomputing.com:1234/?warehouse=test_database');
-
     });
+
+    function assertRepositoryStatus(repositoryId, status) {
+        cy.waitUntil(() =>
+            RepositorySteps.getRepositoryFromList(repositoryId)
+                .should('be.visible')
+                .find('.repository-status .text-secondary').should('contain.text', status));
+    }
+
+    const commonRepositoryParams = {
+        "propertiesFile": {
+            "name": "propertiesFile",
+            "label": "JDBC properties file",
+            "value": "repositories/ontop/ontop_jdbc.properties"
+        },
+        "lensesFile": {
+            "name": "lensesFile",
+            "label": "Lenses file",
+            "value": ""
+        },
+        "isShacl": {
+            "name": "isShacl",
+            "label": "Enable SHACL validation",
+            "value": "false"
+        },
+        "owlFile": {
+            "name": "owlFile",
+            "label": "Ontology file",
+            "value": ""
+        },
+        "member": {
+            "name": "member",
+            "label": "FedX repo members",
+            "value": []
+        },
+        "constraintFile": {
+            "name": "constraintFile",
+            "label": "Constraint file",
+            "value": ""
+        },
+        "obdaFile": {
+            "name": "obdaFile",
+            "label": "OBDA or R2RML file",
+            "value": "repositories/ontop/university-complete.obda"
+        },
+        "dbMetadataFile": {
+            "name": "dbMetadataFile",
+            "label": "DB metadata file",
+            "value": ""
+        }
+    };
+
+    function interceptRepository(id, statusCode, extraParams = {}) {
+        cy.intercept(`rest/repositories/${id}?location=`, {
+            statusCode,
+            body: {
+                "id": id,
+                "title": "",
+                "type": "ontop",
+                "sesameType": "graphdb:OntopRepository",
+                "location": "",
+                "hostName": "localhost",
+                "params": { ...commonRepositoryParams, ...extraParams }
+            }
+        });
+    }
+
+    function stubRepoCreationEndpoints() {
```

Comment:
All stubs needs to be in respective classes here https://github.com/Ontotext-AD/graphdb-workbench/tree/master/test-cypress/stubs